source/NVDAObjects/window/excel.py

```diff
@@ -1517,10 +1527,12 @@ def iterate(self):
 		cellInfos = (ExcelCellInfo * count)()
 		numCellsFetched = ctypes.c_long()
 		address = collectionObject.address(True, True, xlA1, True)
+		sep = worksheet.Application.International(XlApplicationInternational.LIST_SEPARATOR)
+		localeAwareAddress = address.replace(",", sep)
```

Comment:
As the address contains the file name and sheet name,  replacing the commas like this will mangle the file or sheet names if they happen to contain a comma, which Excel definitely allows.
E.g.
```
'[test, ding.xlsx]ding, dong'!$A$1
```
At very least, you need to split on the exclamation mark (!) and only replace the commas after that point.

Reply:
Thanks @michaelDCurran for noting this! I have opened #17754 to fix it up in a more robust way, at least I hope so...